gap/prop.gi

```diff
@@ -555,3 +555,108 @@ function(gr)
   return true;
 
 end);
+
+#
+
```

Comment:
There are no comments in your code here. You should add either a description of the algorithm you are using, or a link/url/reference for the underlying algorithm. Without that this will be harder to maintain this code.